benches/speed_tests.rs

```diff
@@ -396,10 +403,154 @@ pub fn idpf(c: &mut Criterion) {
     }
 }
 
+/// Benchmark Poplar1.
+#[cfg(feature = "experimental")]
+pub fn poplar1(c: &mut Criterion) {
+    let test_sizes = [16_usize, 128, 256];
+
+    let mut group = c.benchmark_group("poplar1_shard");
+    for size in test_sizes.iter() {
+        group.throughput(Throughput::Bytes(*size as u64 / 8));
+        group.bench_with_input(BenchmarkId::from_parameter(size), size, |b, &size| {
+            let bits = iter::repeat_with(random).take(size).collect::<Vec<bool>>();
+            let measurement = IdpfInput::from_bools(&bits);
+            let vdaf = Poplar1::new_aes128(bits.len());
+
+            b.iter(|| {
+                vdaf.shard(&measurement).unwrap();
+            });
+        });
+    }
+    drop(group);
+
+    let mut group = c.benchmark_group("poplar1_prepare_init");
+    for size in test_sizes.iter() {
+        group.throughput(Throughput::Bytes(*size as u64 / 8));
+        group.bench_with_input(BenchmarkId::from_parameter(size), size, |b, &size| {
+            let vdaf = Poplar1::new_aes128(size);
+            let verify_key = random();
+            let nonce: [u8; 16] = random();
+
+            // Parameters are chosen to match Chris Wood's experimental setup:
+            // https://github.com/chris-wood/heavy-hitter-comparison.
+            let (measurements, prefix_tree) = poplar1_generate_zipf_distributed_batch(
+                size, // bits
+                10,   // threshold
+                1000, // number of measurements
+                128,  // Zipf support
+                1.03, // Zipf exponent
```

Comment:
@chris-wood can you take a look at this function to make sure I'm generating the outputs correctly? Also, what do you think would be reasonable worst-case parameters to use here?

Reply:
By correct I mean: Are we generating the measurements in the same way you did in your experiment?

Fair point about the threshold.